righttyper/righttyper_types.py

```diff
@@ -168,4 +168,26 @@ def process(self) -> tuple[TypeInfo, ...]:
             else:
                 retval = TypeInfo("typing", "Generator", (y, TypeInfo("typing", "Any"), self.returns))
 
-        return (*self.args, retval)
+        # This is an aggressive algorithm for transforming types
+        def convert(type: TypeInfo | str):
+            if isinstance(type, str):
+                return type
+            type = TypeInfo(
```

Comment:
You should use `TypeInfo.replace` for things like that.  Within a `TypeInfo.Transformer`...